IanGRap/Squish
# Add new sounds

Contributors: Alfredo, 

Goal: Add the Score Laughing Sound. Add the Transition sound when you click Play and go to the Lobby

Files Changed: 
- VariableSoundPitchShifter.cs: Created a new function that makes it easier to add new sounds groups. Added a sound group for the score Increment laughs and the respawn block break
- AudioManager.cs: Has a new function that will call the ScoreIncrement laugh sounds and the respawn block sounds. The pitch for this group of sounds can be adjusted in the Inspector but I left them with no pitch shift. 
- UIScoreImage.cs: calls the laughing sound when the score is updated but only if the score canvas is active.
- LobbyStateManager.cs: calls play on the transition sound. 
- PlayerRespawn.cs: Added the sound for when a respawn block breaks

**Note**: The new sound sources are not hooked up to the mixer. This should be done later when Fernando is remixing the music. 

How to test: 
* Start in the Lobby Scene. When you click the play button you should hear the transition. Try to spam this button by going into and out of the Lobby. Make sure that this still sounds OK.
* Play through the game until you get the score screen. Make sure that the score increment sound is called at the right time. (right when the score goes up in value)
* While you play make sure that the laugh sound is not played at any other time. I had an issue where the sound was being played even after the score screen was gone because it was updating the numbers in the background.
* Destroy respawn blocks. Both kill a player with one and destroy it normally. 
❤️
--- 
I updated variableSoundPitchShifter.cs with suggestions provided by ian.
